src/bp-core/bp-core-update.php

```diff
@@ -1137,3 +1142,103 @@ function bp_update_to_1_5_5() {
 function bb_update_to_1_5_6() {
 	bp_core_install_moderation_emails();
 }
+
+/**
+ * 1.5.7.2 update routine.
+ *
+ * - Migrate group id to associate forum.
+ * - Update forum meta _bbp_group_ids, associated with group id.
+ *
+ * @since BuddyBoss 1.5.8
+ */
+function bb_update_to_1_5_8() {
+	global $bp_background_updater;
+
+	foreach ( bb_update_to_1_5_8_get_pages() as $page ) {
+		$bp_background_updater->push_to_queue(
+			array(
+				'callback' => 'bb_update_to_1_5_8_migrate_group_id_associate_forum',
+				'args'     => array( 'page' => $page ),
+			)
+		);
+	}
+	
+	$bp_background_updater->save()->schedule_event();
+}
+
+/**
+ * Chunk the total groups as pages.
+ *
+ * @since BuddyBoss 1.5.8
+ *
+ * @return array
+ */
+function bb_update_to_1_5_8_get_pages() {
+	$total_groups = groups_get_total_group_count();
```

Comment:
please check groups component is active or not before using functions from group component, otherwise it will throw fatal error for site which do not have enabled groups component.